frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx

```diff
@@ -14,87 +28,218 @@ const TABS = {
   Appearance: "appearance",
 };
 
-type StaticEmbedSetupPaneProps = OverviewSettingsProps &
-  ParametersSettingsProps &
-  AppearanceSettingsProps;
+export interface StaticEmbedSetupPaneProps {
+  resource: EmbedResource;
+  resourceType: EmbedResourceType;
+  resourceParameters: EmbedResourceParameter[];
+
+  onUpdateEnableEmbedding: (enableEmbedding: boolean) => void | Promise<void>;
+  onUpdateEmbeddingParams: (
+    embeddingParams: EmbeddingParameters,
+  ) => void | Promise<void>;
+}
 
 export const StaticEmbedSetupPane = ({
-  activePane,
   resource,
   resourceType,
-  embedType,
-  token,
-  iframeUrl,
-  siteUrl,
-  secretKey,
-  params,
-  displayOptions,
-  lockedParameters,
-  parameterValues,
   resourceParameters,
-  embeddingParams,
-  onChangeDisplayOptions,
-  onChangeEmbeddingParameters,
-  onChangeParameterValue,
-  onChangePane,
+  onUpdateEnableEmbedding,
+  onUpdateEmbeddingParams,
 }: StaticEmbedSetupPaneProps): JSX.Element => {
+  const [activePane, setActivePane] = useState<ActivePreviewPane>("code");
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
+  const secretKey = checkNotNull(
```

Comment:
I saw that these 2 props are being passed through a bunch of components, what about we declare these states at the leaf component where they're used? That should probably help with the props drilling. What do yuo think?